src/main/java/frc/robot/subsystems/arm/ArmAngleSubsystem.java

```diff
@@ -0,0 +1,139 @@
+
+        m_armAngleFollower.follow(m_armAngleMaster);
+    
+        m_encoderArmAngle = new DutyCycleEncoder(ArmConstants.ENCODER_PORT);
+        m_encoderArmAngle.reset();
+        m_encoderArmAngle.setDistancePerRotation(360);
+        m_encoderArmAngle.setPositionOffset(0);
+
+        m_anglePID = new PIDController(ArmConstants.KP_ANGLE, ArmConstants.KI_ANGLE, 0.0);
+        m_anglePID.setTolerance(3);
+
+        m_inputs = new ArmAngleIOInputsAutoLogged();
+
+        armAngleTab = Shuffleboard.getTab("ArmAngleSubsystem");
+    }
+
+    @Override
+    public void periodic() {
+        updateInputs(m_inputs);
+        Logger.getInstance().processInputs("Arm Angle", m_inputs);
+
+        // Booleans
+        // Misc.
+        armAngleTab.add("Arm Angle At setpoint", armAngleAtSetpoint());
+        armAngleTab.add("Encoder connected", encoderConnected());
+        // Limits
+        armAngleTab.add("At upper limit", armAtUpperLimit());
+        armAngleTab.add("At lower limit", armAtLowerLimit());
+        // Motor inversions
+        armAngleTab.add("Master inverted", m_armAngleMaster.getInverted());
+        armAngleTab.add("Follower Inverted", m_armAngleFollower.getInverted());
+
+        // Doubles
+        // Angles
+//        armAngleTab.add("Encoder raw", m_encoderArmAngle);
+        armAngleTab.add("Angle raw", m_encoderArmAngle.getAbsolutePosition() * 360);
+        armAngleTab.add("Angle converted", getArmAngle());
+        // Targets
+        armAngleTab.add("Target", prevSetpointRaw);
+        armAngleTab.add("Clamped setpoint", prevSetpointClamped);
+        armAngleTab.add("PID setpoint output", prevSetpointPID);
+    }
+
+    public void updateInputs(ArmAngleIOInputsAutoLogged inputs){
+        inputs.armAngle = getArmAngle();
+    }
+
+    public void setAngleSpeed(double speed) {
+        if ((getArmAngle() <= ArmConstants.K_REVERSE_LIMIT && speed <= 0)
+            || (getArmAngle() >= ArmConstants.K_FORWARD_LIMIT && speed >= 0)) {
+            m_armAngleMaster.set(speed);
+        }
+    }
+
+    public Command setAngleSpeedFactory(double speed) {
+        return runOnce(() -> setAngleSpeed(speed));
+    }
+
+    public void setArmAngle(double targetAngleRaw){
+        // Get angle
+        double currentArmAngle = getArmAngle();
+
+        // Clamp target
+        double targetAngleClamped = MathUtil.clamp(targetAngleRaw, ArmConstants.K_REVERSE_LIMIT, ArmConstants.K_FORWARD_LIMIT);
+        double targetAnglePID = MathUtil.clamp(m_anglePID.calculate(currentArmAngle, targetAngleClamped), -6, 6);
+
+        // Update dashboard variables
+        prevSetpointRaw = targetAngleRaw;
+        prevSetpointClamped = targetAngleClamped;
+        prevSetpointPID = targetAnglePID;
+
+        // Set voltage based off of PID
+        m_armAngleMaster.setVoltage(-targetAnglePID);
+    }
+
+    public double getArmAngle() {
+        return Utils.normalize((m_encoderArmAngle.getAbsolutePosition() * 360) - ArmConstants.ARM_OFFSET);
```

Comment:
Refactor to return ~30 degrees for stow, and ~300 degrees for intake

Reply:
We'll have to do this once the robot is toghether, as the offset is hardware specific